From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Date: Tue, 20 Aug 2019 16:57:32 +0300
Subject: [PATCH] Revert "xen/arm: don't iomem_permit_access for
 reserved-memory regions"

This reverts commit f9335bd4b89ba398ae993a0655c40c545b75d7c1
(manual revert).

Signed-off-by: Oleksandr Tyshchenko <oleksandr_tyshchenko@epam.com>
Signed-off-by: Dmytro Firsov <dmytro_firsov@epam.com>
(cherry picked from commit 5896fb9c309872b651e177b80fecdb7a4e9f45ec)
---
 xen/arch/arm/device.c | 25 ++++++++-----------------
 1 file changed, 8 insertions(+), 17 deletions(-)

diff --git a/xen/arch/arm/device.c b/xen/arch/arm/device.c
index 7d9644fd8b..8d0c6196f9 100644
--- a/xen/arch/arm/device.c
+++ b/xen/arch/arm/device.c
@@ -64,24 +64,15 @@ int __overlay_init map_range_to_domain(const struct dt_device_node *dev,
         return -ERANGE;
     }
 
-    /*
-     * reserved-memory regions are RAM carved out for a special purpose.
-     * They are not MMIO and therefore a domain should not be able to
-     * manage them via the IOMEM interface.
-     */
-    if ( strncasecmp(dt_node_full_name(dev), "/reserved-memory/",
-                     strlen("/reserved-memory/")) != 0 )
+    res = iomem_permit_access(d, paddr_to_pfn(addr),
+            paddr_to_pfn(PAGE_ALIGN(addr + len - 1)));
+    if ( res )
     {
-        res = iomem_permit_access(d, paddr_to_pfn(addr),
-                                  paddr_to_pfn(addr + len - 1));
-        if ( res )
-        {
-            printk(XENLOG_ERR "Unable to permit to dom%d access to"
-                    " 0x%"PRIx64" - 0x%"PRIx64"\n",
-                    d->domain_id,
-                    addr & PAGE_MASK, PAGE_ALIGN(addr + len) - 1);
-            return res;
-        }
+        printk(XENLOG_ERR "Unable to permit to dom%d access to"
+                " 0x%"PRIx64" - 0x%"PRIx64"\n",
+                d->domain_id,
+                addr & PAGE_MASK, PAGE_ALIGN(addr + len) - 1);
+        return res;
     }
 
     if ( !mr_data->skip_mapping )
-- 
2.43.0

